deps/parser/parser.jl

```diff
@@ -0,0 +1,184 @@
+    return $(typestring)call_method( "$polymake_name$parameter_string", dispatch_obj, Array{Any,1}([ $call_args ]) )
+end
+
+"""
+end
+
+function create_argument_string_from_type( type_string::String, number::Int64 )
+    ## Special case for BigObject (we do not care 'bout the type here)
+    if startswith(type_string,"BigObject")
+        return "arg" * string(number) * "::" * type_translate_list["BigObject"]
+    end
+    ## We only translate the known cases
+    if !haskey(type_translate_list,type_string)
+        println("Unknown type: " * type_string)
+        return false
+    end
+    return "arg" * string(number) * "::" * type_translate_list[type_string]
+end
+
+function parse_function_definition(method_dict::Dict, app_name::String)
+    name = method_dict["name"]
+    arguments = method_dict["args"]
+    mandatory = method_dict["mandatory"]
+    type_params = method_dict["type_params"]
+    is_method = haskey(method_dict,"method_of")
+    ## for now, we use the same name for Julia and Polymake
+    julia_name = name
+    polymake_name = name
+
+    ## Check for option set
+    has_option_set = last(arguments) == "OptionSet"
+
+    ## Make type parameters
+    param_list_header = Array{String,1}()
+    param_list = Array{String,1}()
+    for i in 1:type_params
+        push!(param_list_header,"param$i::String")
+        push!(param_list,"\$param$i")
+    end
+    if length(param_list) >= 1
+        parameter_string = "<" * join(param_list,",") * ">"
+    else
+        parameter_string = ""
+    end
+
+    ## Compute the argument range
+    max_argument_number = length(arguments) - (has_option_set ? 1 : 0)
+    min_argument_number = mandatory
+
+    ## Option set parameter
+    option_set_argument = "option_set::" * type_translate_list["OptionSet"]
+    option_set_parameter = "option_set"
+
+    ## Real arguments
+    if is_method
+        argument_list_header = Array{String,1}(["dispatch_obj::$(typestring)pm_perl_Object"])
+        ## Fixme: good?
+        min_argument_number = min_argument_number + 1
+        max_argument_number = max_argument_number + 1
+    else
+        argument_list_header = Array{String,1}()
+    end
+    argument_list = Array{String,1}()
+
+    for i in 1:max_argument_number
+        push!(argument_list_header,create_argument_string_from_type(arguments[i],i))
+        push!(argument_list,"arg$i")
+    end
+
+    if false in argument_list_header
+        for i in 1:length(argument_list_header)
+            if argument_list_header[i] == false
+                if i <= min_argument_number
+                    println("Cannot parse function " * name * " due to unparsable argument")
+                    return false
+                else
+                    println("Ignoring " * string(max_argument_number-i) * " arguments of function " * name)
+                    max_argument_number = i - 1
+                end
+            end
+        end
+    end
+
+    return_string = ""
```

Comment:
this is probably the place to separate json parsing from code generation

Reply:
Please elaborate?

Reply:
below we mostly fill the return_string with all the data we pulled (and transformed) from json;
and the name of the function reads `parse...`, whereas it does both parsing and code gen

Reply:
K.

Reply:
I decided to not do this at the moment. The name of the function in the current state is probably debatable, but currently there is really no use in ripping this function apart into two.

Granted, two would be a bit more clean, but you end up in having to parse all the extracted information from the first part to the second part. Either using keyword arguments or a dict, since everything else would be very messy.

But since both parts are resonably small, I would leave them as they are.